packages/react-storage/src/components/StorageBrowser/actions/handlers/download.ts

```diff
@@ -1,16 +1,59 @@
-import { TaskHandler, TaskHandlerInput, TaskHandlerOutput } from '../types';
+import { getUrl } from '../../storage-internal';
+import {
+  TaskHandler,
+  TaskHandlerInput,
+  TaskHandlerOptions,
+  TaskHandlerOutput,
+} from '../types';
+import { constructBucket, resolveHandlerResult } from './utils';
 
-interface DownloadHandlerData {
-  key: string;
-}
-interface DownloadHandlerOptions {
-  onProgress?: (progress: number) => void;
-}
+interface DownloadHandlerOptions extends TaskHandlerOptions {}
 export interface DownloadHandlerInput
-  extends TaskHandlerInput<DownloadHandlerData, DownloadHandlerOptions> {}
+  extends Omit<TaskHandlerInput<never, DownloadHandlerOptions>, 'data'> {
+  data: { key: string };
+}
 export interface DownloadHandlerOutput extends TaskHandlerOutput {}
 
 export interface DownloadHandler extends TaskHandler<DownloadHandlerInput> {}
 
-export const downloadHandler: DownloadHandler =
-  null as unknown as DownloadHandler;
+function downloadFromUrl(fileName: string, url: string) {
+  const a = document.createElement('a');
+
+  a.href = url;
+  a.download = fileName;
+
+  document.body.appendChild(a);
+
+  a.click();
+
+  document.body.removeChild(a);
+}
+
+export const downloadHandler: DownloadHandler = ({
+  config,
+  data: { key },
+  prefix,
+  options,
+}): DownloadHandlerOutput => {
+  const { credentials } = config;
+  const bucket = constructBucket(config);
+  const fileKey = `${prefix}${key}`;
+
+  const result = getUrl({
+    path: fileKey,
+    options: {
+      bucket,
+      locationCredentialsProvider: credentials,
+      validateObjectExistence: true,
+      contentDisposition: 'attachment',
+    },
+  }).then((result) => {
+    downloadFromUrl(fileKey, result.url.toString());
+    return result;
```

Comment:
returning the actual getUrl return here to preserve the actual output of getUrl. Dont think it is useful anyway though but not sure. If we prefer just leaving it as `Promise<void>` I can pull this lien out.